File: Semestrul 4/MPP2/Problema8_FX/src/main/java/ro/mpp2025/problema8_fx/Domain/Child.java
package ro.mpp2025.problema8_fx.Domain;

import ro.mpp2025.problema8_fx.Utils.AgeConverter;

import java.util.Objects;

public class Child extends Entity<Long> {

    private String name;
    private String CNP;

    public Child(Long aLong, String name, String CNP) {
        setId(aLong);
        this.name = name;
        this.CNP = CNP;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCNP() {
        return CNP;
    }

    public void setCNP(String CNP) {
        this.CNP = CNP;
    }

    public Integer getAge() {
        return AgeConverter.getAgeFromCNP(CNP);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Child child = (Child) o;
        return Objects.equals(name, child.name) && Objects.equals(CNP, child.CNP);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, CNP);
    }

    @Override
    public String toString() {
        return "Child{" +
                "name='" + name + '\'' +
                ", CNP='" + CNP + '\'' +
                '}';
    }
}

File: Semestrul 4/MPP2/Problema8_FX_Csharp/Problema8_FX_Csharp/Domain/Child.cs
using Problema8_FX_Csharp.Utils;

namespace Problema8_FX_Csharp.Domain;

public class Child : Entity<long>
{
    public String Name { get; set; }
    public String CNP { get; set; }
    
    public Child(long id, String name, String cnp)
    {
        SetId(id);
        Name = name;
        CNP = cnp;
    }

    public int getAge()
    {
        return AgeConverter.getAgeFromCNP(CNP);
    }
    
    public override String ToString()
    {
        return $"Name: {Name}, CNP: {CNP}";
    }
    
    public override bool Equals(Object obj)
    {
        if (obj == null || GetType() != obj.GetType())
        {
            return false;
        }
        
        Child child = (Child)obj;
        return Name.Equals(child.Name) && CNP.Equals(child.CNP);
    }
}